model/flow/event.go

```diff
@@ -99,18 +99,28 @@ type BlockEvents struct {
 
 type EventsList []Event
 
-// EventsListHash calculates a hash of events list,
-// by simply hashing byte representation of events in the lists
-func EventsListHash(el EventsList) (Identifier, error) {
-
-	hasher := hash.NewSHA3_256()
+// EventsListMerkleHash calculates the root hash of events inserted into a
+// merkle trie with the hash of event as the key and encoded event as value
+func EventsListMerkleHash(el EventsList) (Identifier, error) {
```

Comment:
I think the usage of the work `List` in the old function was there to indicate that the hashed object is a list. In the new code, the object that we hash is a tree, not a list anymore. Hence, I would suggest to drop the `List` from the function:
```suggestion
func EventsMerkleRootHash(el EventsList) (Identifier, error) {
```

Reply:
renamed to EventsMerkleRootHash